dandiapi/api/services/asset/__init__.py

```diff
@@ -44,6 +44,39 @@ def _create_asset(
     return asset
 
 
+def _add_asset_to_version(
+    *,
+    version: Version,
+    asset_blob: AssetBlob | None,
+    zarr_archive: ZarrArchive | None,
+    metadata: dict,
+) -> Asset:
+    path = metadata['path']
+    asset = _create_asset(
+        path=path, asset_blob=asset_blob, zarr_archive=zarr_archive, metadata=metadata
+    )
+    version.assets.add(asset)
+    add_asset_paths(asset, version)
+
+    # Trigger a version metadata validation, as saving the version might change the metadata
+    version.status = Version.Status.PENDING
+    # Save the version so that the modified field is updated
+    version.save()
```

Comment:
Since the version isn't necessarily locked here, it might be unsafe to call .save() on it like this. Couldn't this just be

```suggestion
    Version.objects.filter(id=version.id).update(status=Version.Status.PENDING, modified=timezone.now())
```